### DISCUSSION TITLE
Is there a simple way to manage pypi repositories, a-la Poetry's `poetry config repositories.name https://url/` ?

### DISCUSSION THREAD

**Discussion by redjax** [pdm-project/pdm#2117] [Q&A] [answered]
Before going into this question, I have searched discussions for adding repos, and it seems like it's just the same solutions as the PDM documentation lists. My question is along the same lines, but might not be identical, so I wanted to post this anyway.

I recently started building my first Python package (with PDM, of course!), and ran into a lot of issues publishing it. I ended up making an entry in `~/.pypirc` for pypi and test.pypi, added my username as `__token__` and password as my API token, and I use Twine to push releases.

This setup process is clunky when working on different machines, but not a huge deal. However, I went back to Poetry briefly just to try the publishing process...and found it to be so much simpler and easy to navigate vs PDM.

This might just be me being new to the packaging process, but the process of adding a repository with Poetry's `poetry config repositories.name <url>`, then setting a token with `poetry config name.token <token>` and then just being able to run `poetry publish` has been huge.

Is there a simple way to configure alternate pypi repositories with PDM? I've read the documentation for this process, it sounds like editing the global config.toml file is the only (obvious/documented) way to do this, and if that's the case, I don't think this one little nicety is enough to pull me back over to Poetry; I'm very happy with PDM and plan to use it for the foreseeable future. But being able to publish in a simpler manner would be very nice :)

**Comment by frostming** [comment, marked as answer]
If you read the docs, why did you miss this part? https://pdm.fming.dev/latest/usage/config/#configure-the-repositories-for-upload:~:text=To%20change%20the%20repository%20config%20from%20the%20command%20line%2C%20use%20the%20pdm%20config%20command%3A

**Reply by redjax** [reply]
I just tried this again on my home computer, and it actually worked the way I expected...I'm not sure if I was maybe looking at a different part of the docs, but thank you for the direct link!

When I did this on my work laptop earlier, I *must* have been typing the commands wrong. I was getting an error about PDM being unable to find a repository by name `pypi-test`. I'm sorry, I don't have a screenshot of the error, but it doesn't matter at this point because what you linked worked!

Thank you for creating such a great tool :) Sorry for the dumb question, but thank you for your quick help. I'm very happy it's this simple on PDM and I was just doing something wrong 🤘
